pkg/operation/botanist/component/kubeapiserverexposure/kube_apiserver_service.go

```diff
@@ -150,6 +154,11 @@ func (s *service) Deploy(ctx context.Context) error {
 			metav1.SetMetaDataAnnotation(&obj.ObjectMeta, "networking.istio.io/exportTo", "*")
 		}
 
+		utilruntime.Must(gardenerutils.InjectNetworkPolicyAnnotationsForScrapeTargets(obj,
+			networkingv1.NetworkPolicyPort{Port: utils.IntStrPtrFromInt(kubeapiserver.Port), Protocol: utils.ProtocolPtr(corev1.ProtocolTCP)},
+			networkingv1.NetworkPolicyPort{Port: utils.IntStrPtrFromInt(monitoring.BlackboxExporterPort), Protocol: utils.ProtocolPtr(corev1.ProtocolTCP)},
```

Comment:
Is this still required? This is the port of the blackbox sidecar of the Prometheus pod. I tried to remove it in my tests and the target could still be scraped.
/cc @istvanballok @rickardsjp

Reply:
Very nice catch :) Indeed, this was a left-over. With https://github.com/gardener/gardener/pull/91, a `blackbox-exporter` sidecar was introduced to the `kube-apiserver` pod. However, this was removed with https://github.com/gardener/gardener/pull/2267, but apparently this PR missed to adapted the `NetworkPolicy` rules.
Removed now, thanks!